wagtail/admin/views/pages/listing.py

```diff
@@ -20,13 +28,33 @@
     PageTable,
     PageTitleColumn,
 )
-from wagtail.admin.views.generic.base import BaseListingView
-from wagtail.admin.views.generic.permissions import PermissionCheckedMixin
-from wagtail.models import Page
+from wagtail.admin.views.generic.models import IndexView as GenericIndexView
+from wagtail.models import Page, get_page_models
 from wagtail.permissions import page_permission_policy
 
 
-class BaseIndexView(PermissionCheckedMixin, BaseListingView):
+def get_content_types_for_filter():
+    models = [model.__name__.lower() for model in get_page_models()]
+    return ContentType.objects.filter(model__in=models).order_by("model")
+
+
+class PageFilterSet(WagtailFilterSet):
+    content_type = MultipleContentTypeFilter(
+        label=_("Page type"),
+        queryset=lambda request: get_content_types_for_filter(),
+        widget=CheckboxSelectMultiple,
+    )
+    latest_revision_created_at = DateFromToRangeFilter(
+        label=_("Date Updated"),
+        widget=DateRangePickerWidget,
+    )
+
+    class Meta:
+        model = Page
+        fields = ["content_type"]
```

Comment:
I was curious why the `latest_revision_created_at` works without specifying it here, but it looks like you only need to specify the fields that you want to be automatically generated (like `ModelForm`).

However, we can't leave out the `fields` definition, so this should probably be

```suggestion
        fields = []
```

or include both fields if we want to be explicit. Maybe add a comment to explain?